File: scala-spring/commandside-backend-accounts/src/main/scala/net/chrisrichardson/eventstore/examples/bank/accounts/AccountConfiguration.scala
package net.chrisrichardson.eventstore.examples.bank.accounts

import net.chrisrichardson.eventstore.EventStore
import net.chrisrichardson.eventstore.subscriptions.EnableEventHandlers
import net.chrisrichardson.utils.config.MetricRegistryConfiguration
import org.springframework.context.annotation.{Bean, Configuration, Import}

@Configuration
@Import(Array(classOf[MetricRegistryConfiguration]))
@EnableEventHandlers
class AccountConfiguration {

  @Bean
  def accountService(eventStore : EventStore) = new AccountService()(eventStore)


  @Bean
  def transferWorkflow(eventStore: EventStore): TransferWorkflowAccountHandlers = {
    new TransferWorkflowAccountHandlers(eventStore)
  }
}

File: java-spring/commandside-backend-accounts/src/main/java/net/chrisrichardson/eventstore/javaexamples/banking/backend/commandside/accounts/AccountConfiguration.java
package net.chrisrichardson.eventstore.javaexamples.banking.backend.commandside.accounts;

import net.chrisrichardson.eventstore.EventStore;
import net.chrisrichardson.eventstore.repository.AggregateRepository;
import net.chrisrichardson.eventstore.subscriptions.EnableEventHandlers;
import net.chrisrichardson.eventstore.subscriptions.EventHandlerRegistrarFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
@EnableEventHandlers
public class AccountConfiguration {


  @Autowired
  private EventHandlerRegistrarFactory eventHandlerRegistrarFactory;

  @Bean
  public AccountWorkflow accountWorkflow() {
    return new AccountWorkflow();
  }


  @Bean
  public AccountService accountService(AggregateRepository<Account, AccountCommand> accountRepository) {
    return new AccountService(accountRepository);
  }

  @Bean
  public AggregateRepository<Account, AccountCommand> accountRepository(EventStore eventStore) {
    return new AggregateRepository<Account, AccountCommand>(Account.class, eventStore);
  }

}


